tests/test_downloadermiddleware_cookies.py

```diff
@@ -229,3 +229,17 @@ def test_local_domain(self):
         assert self.mw.process_request(request, self.spider) is None
         self.assertIn('Cookie', request.headers)
         self.assertEqual(b'currencyCookie=USD', request.headers['Cookie'])
+
+    def test_keep_cookie_header(self):
```

Comment:
could you please also add a test which checks that it works with DEFAULT_HEADERS_MIDDLEWARE?